### PR TITLE
Modeling

### PR DESCRIPTION
**Pull request by malarzm**
Modeling part of the library:
- skeleton for Client
- and all value objects we'll be supporting

### CODE REVIEW

File: src/ValueObject/PUDO.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+{
+    public string $id;
+    public bool $active;
+    public Type $type;
+    public string $language;
+    public Address $address;
+    public Coordinates $coordinates;
+    public AdditionalInfo $additionalInfo;
+    /**
+     * @var OpeningTime[]
```
Comment (malarzm) [inline src/ValueObject/PUDO.php:19]: I'm thinking about some `Week` concept instead of a plain array

File: src/Enum/Type.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace Answear\DpdPlPickupServicesBundle\Enum;
+
+use MabeEnum\Enum;
+
+class Type extends Enum
+{
+    public const STANDARD = 100;
+    public const NETWORK = 200;
+    public const FULL_TIME = 300;
```
Comment (malarzm) [inline src/Enum/Type.php:13]: Not sure about naming for the last two: these are "punkt sieciowy" and "punkt etatowy" in Polish, I'm on a look out for better translations

**Comment by konradkozaczenko** [comment]
Don't we want Enum suffix for class names in Enum namespace? Now, you have Service class in Enum namespace, Service namespace and ServiceException class and it seems to me a little bit confusing

File: src/Enum/Type.php (hunk 2 above)
Comment (lukasz-falda) [reply src/Enum/Type.php:13]: "punkt sieciowy" is not something like "chain of point"? Similar to `chain of stores`?
Of course it is guesswork

File: src/Service/PUDOList.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    public function __construct(ConfigProvider $configProvider, ?ClientInterface $client = null)
+    {
+        $this->configProvider = $configProvider;
+        $this->client = $client ?? new Client(
+                [
+                    'base_uri' => $configProvider->getUrl(),
+                    'http_errors' => false,
+                    'timeout' => $configProvider->getRequestTimeout(),
+                ]
+            );
```
Comment (lukasz-falda) [inline src/Service/PUDOList.php:26]: are you sure you want to switch off http_errors?

Comment (malarzm) [reply src/Service/PUDOList.php:26]: Errors from DPD looks like this:
```
<RESPONSE><ERRORcode="306"><VALUE>Pole max_pudo_number przyjmuje wartości całkowitoliczbowe z przedziału od 1 do 25.</VALUE></ERROR></RESPONSE>
```
and from what I've seen authorization gets 4xx. The idea is that each such response will be turned into an exception. If no proper XML is returned there'll be another exception.

**Comment by malarzm** [comment]
> Don't we want Enum suffix for class names in Enum namespace?

I don't :see_no_evil: but if everybody wants them I'll add them

**Review by konradkozaczenko — APPROVED** [review]

File: src/Enum/Type.php (hunk 2 above)
Comment (malarzm) [reply src/Enum/Type.php:13]: Changed in e5561fea90955d7b53dfdc544d29d0388d111489

File: src/ValueObject/PUDO.php (hunk 1 above)
Comment (malarzm) [reply src/ValueObject/PUDO.php:19]: Introduced in 2bf33a77984e7b14727d7e630ac9ccd861065fa6

**Pull request closed without merging**